src/main/services/fetchProxy.ts

```diff
@@ -0,0 +1,10 @@
+import fetch, { RequestInfo, RequestInit } from "node-fetch";
+import httpsProxyAgent from "https-proxy-agent";
+import { SYS_SETTINGS } from "../../index";
+
+export default function fetchProxy(url: RequestInfo, init: RequestInit = {}) {
+  if (SYS_SETTINGS.proxy) {
+    init.agent = httpsProxyAgent(SYS_SETTINGS.proxy);
+  }
+  return fetch(url, init);
+}
```

Comment:
```suggestion
const fetchProxy = (url: RequestInfo, init: RequestInit = {}) => {
  const proxy = process.env.HTTPS_PROXY || process.env.HTTP_PROXY || SYS_SETTINGS.proxy;
  if (proxy) {
    init.agent = httpsProxyAgent(SYS_SETTINGS.proxy);
  }
  return fetch(url, init);
}

export default fetchProxy;
```

I made a final test and changing your code like this make the feature work out-of-the-box for most people 🎉